src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/DataImportService.cs

```diff
@@ -148,5 +151,63 @@ public async Task UpdateStatus(Guid id, DataImportStatus newStatus, double perce
             context.DataImports.Update(import);
             await context.SaveChangesAsync();
         }
+
+        public async Task WriteDataSetFileMeta(Guid subjectId)
+        {
+            await using var contentDbContext = _dbContextSupplier.CreateDbContext<ContentDbContext>();
+            await using var statisticsDbContext = _dbContextSupplier.CreateDbContext<StatisticsDbContext>();
+
+            var observations = await statisticsDbContext.Observation
+                .AsNoTracking()
+                .Where(o => o.SubjectId == subjectId)
+                .Select(o => new { o.Location.GeographicLevel, o.Year, o.TimeIdentifier })
+                .Distinct()
+                .ToListAsync();
+
+            var geographicLevels = observations
+                .Select(o => o.GeographicLevel.GetEnumLabel())
+                .Distinct()
+                .OrderBy(gl => gl)
```

Comment:
Can this be done with `Sort()`?   Not suggesting it's worth changing - just curious!

Reply:
Seems not. At first I thought it was just queryables didn't have `Sort`, but seems it also isn't a thing if we materialise the query. It seems `Sort` is a `List` thing that doesn't return a new `List`, but sorts the list it has been called on. So to use it in this situation we'd have to do:
```
var geographicLevels = observations
                .Select(o => o.Location.GeographicLevel.GetEnumLabel())
                .Distinct()
                .ToList();
geographicLevels.Sort();
```

Reply:
Ahh yeah of course.  Fair enough, thanks for checking!